matlab/CMakeLists.txt

```diff
@@ -131,6 +131,14 @@ if(NOT Arrow_FOUND)
   build_arrow()
 endif()
 
+# Identify MATLAB build mode (Debug or Release)
+if(WIN32)
+  execute_process(COMMAND dumpbin /dependents ${Matlab_MAIN_PROGRAM} OUTPUT_VARIABLE DEPENDENTS)
```

Comment:
@kevingurney Is there any MATLAB main program that is built in Debug mode on Windows?

If all MATLAB main programs are built in Release mode on Windows, we don't need to check this.

We just need to check our build mode instead:

```cmake
if(WIN32)
  string(TOUPPER "${CMAKE_BUILD_TYPE}" CMAKE_BUILD_TYPE_UPPER)
  if(CMAKE_BUILD_TYPE_UPPER STREQUAL DEBUG)
    message(FATAL_ERROR "...")
  endif()
endif()
```

Reply:
Should I resolve the changes? or wait for @kevingurney  confirmation?

Reply:
> @kevingurney Is there any MATLAB main program that is built in Debug mode on Windows?
> 
> If all MATLAB main programs are built in Release mode on Windows, we don't need to check this.
> 
> We just need to check our build mode instead:
> 
> ```cmake
> if(WIN32)
>   string(TOUPPER "${CMAKE_BUILD_TYPE}" CMAKE_BUILD_TYPE_UPPER)
>   if(CMAKE_BUILD_TYPE_UPPER STREQUAL DEBUG)
>     message(FATAL_ERROR "...")
>   endif()
> endif()
> ```

@kou Should I start working upon this change?

Reply:
Can we wait for a response from @kevingurney (or @sgilmore10 / @leihou6116 may know)?

Reply:
Yaa sure :))

Reply:
@kou and @ShaiviAgarwal2 - sorry for delay!

To answer your question: all publicly released versions of MATLAB are built in release mode (i.e. linked against a release version of the MSVC Runtime Libraries).